### ISSUE TITLE
InfluxDB v0.9 support

### ISSUE THREAD

**Issue by benfb** [google/cadvisor#743]
Are there any plans to support the new InfluxDB version? 0.9 isn't officially out yet, but it seems like it should be done soon, and will be the first stable production release. Unfortunately the API that cadvisor uses to connect  has changed drastically since the last supported version, 0.8.

**Comment by rjnagal** [comment]
We just started looking into it. Thanks for reporting the issue. 

@anushree-n we will probably need to consider both new and old influxdb API while updating the metrics  schema.

**Comment by mnuessler** [comment]
@rjnagal I'm very interested in support for InfluxDB 0.9 too, so I gave it a try last week and implemented a prototype that can already write stats to InfluxDB 0.9. It is not ready to be merged yet, so I did not open a PR yet. (I still have to fix the *RecentStats()* function and the tests.)

I think I could finish this up during the next few days. In case you would like to have a look at the code already, please refer to [this branch](https://github.com/egymgmbh/cadvisor/tree/influx09).

**Comment by rjnagal** [comment]
Thanks @mnuessler for taking it up! It can wait a few days.

We are in process of removing RecentStats() as we don't use it anymore. I'll send a PR for removing it today.

**Comment by neonstalwart** [comment]
fyi - InfluxDB 0.9 is still experiencing some API churn.  be sure to target the [alpha1](https://github.com/influxdb/influxdb/tree/alpha1) branch (if it has not been merged to master).  that merge is anticipated to happen some time later this week.

**Comment by mnuessler** [comment]
Thanks for the info! With RecentStats() removed there is less work left to do for me.

So far I have tested against the API of 0.9.0-rc31, but I will make sure to test against the alpha1 branch also. Good point!

**Comment by benfb** [comment]
InfluxDB v0.9 is officially released now, see here: http://influxdb.com/blog/2015/06/11/InfluxDB-v0_9_0-released-with-developer-and-production-support.html

The release notes claim "going forward in the 0.9.x series of releases we will not make breaking API changes or breaking changes to the underlying data storage."

**Comment by mnuessler** [comment]
@benfb That's great news! I will test against the 0.9.0 release version.

**Comment by SlashmanX** [comment]
Has there been any movement on this?

**Comment by mnuessler** [comment]
I think I will be able to open a PR tomorrow. I had to change a lot when upgrading from  0.9.0-rc31 to the 0.9.0 release version.

**Comment by vmarmol** [comment]
Eager to see that @mnuessler :)

**Comment by jmaitrehenry** [comment]
@mnuessler if you need some help, I'm in :)

**Comment by mnuessler** [comment]
@jmaitrehenry Thanks a lot :) I have just opened the PR, but I'm sure there are still many improvements to make!

**Comment by hectorj2f** [comment]
Are there any expectations of having this PR merged soon ?

**Comment by vmarmol** [comment]
Sorry, we've been swamped this week with the Kubernetes 1.0 launch. Will try to take a look today/tomorrow. Nothing against the PR, have just been swamped :)

**Comment by hectorj2f** [comment]
Great! :+1:

**Comment by martinm82** [comment]
Hi guys, just wondering whether the PR will be merged into master soon? :)

**Comment by jwthomp** [comment]
Also following up to check on status.

**Comment by ClanStyles** [comment]
Any updates, this is one of the only reasons I haven't upgraded influxdb.

**Comment by selvik** [comment]
Looking forward to the merge of this in to master. :) Given that the golang client now works with only InfluxDB 0.9, this merge would be very helpful for us! Thank you!

**Comment by rjnagal** [comment]
Looks like influxdb broke the older version, so we can ignore that too.

**Comment by harunyardimci** [comment]
Hi, 

is there any due date for this?

**Comment by hkeeler** [comment]
Any update on this?

**Comment by aheusingfeld** [comment]
> Looks like influxdb broke the older version, so we can ignore that too.

@rjnagal I'm not sure I get what you meant to say. The InfluxDB API shouldn't have changed since 0.9. Could you please have another look at this and explain what is missing from being merged? We're really looking forward to the InfluxDB 0.9 support. Thanks for your time!

**Comment by leonty** [comment]
Hi guys, any observable release date for this PR?
I'm looking forward supporing Influx 0.9 because my infrastructure already uses this version and I can't thus integrace cAdvisor into it :(

**Comment by svenmueller** [comment]
Hi...what is the current schedule for merging the PR?

**Comment by vishh** [comment]
We are waiting on InfluxDB client libraries to be cleaned up in their upcoming v0.9.5 release.

**Comment by ChamaraG** [comment]
Hi, any update on this? Thanks!

**Comment by svenmueller** [comment]
+1

**Comment by martinm82** [comment]
+1

**Comment by bill0ute** [comment]
+1

**Comment by jvgutierrez** [comment]
Any updates on this?

**Comment by jimmidyson** [comment]
There is a PR (#800) in for this but needs updating now that 0.9.5 of influxdb has been released with fewer client deps. Waiting on the contributor to update. If we don't get an update soon-ish, then we will be looking for someone else to pick up this contribution so volunteers welcome.

**Comment by carmark** [comment]
@jimmidyson 

I can take this if the contributor is busy on more important issues.

**Comment by jimmidyson** [comment]
@carmark Would you be able to ask the contributor on #800 if they'd like some help?

**Comment by carmark** [comment]
@jimmidyson  sure.

**Comment by thanhson1085** [comment]
+1

**Comment by jcoloh** [comment]
+1

**Comment by sensimilla** [comment]
+1

**Comment by bmlynarczyk** [comment]
+1

**Comment by asa511** [comment]
+1

**Comment by ohmystack** [comment]
:thumbsup:

**Comment by marcellodesales** [comment]
:+1: @vmarmol Happy New Year!!! It is a promising 2016 and a lot of people is waiting for this to be merged... 6 months ago you mentioned you would be taking a look at it... What needs to be reviewed? Can someone on this thread help?
